src/observer-locator.js

```diff
@@ -188,10 +188,11 @@ export class ObserverLocator {
         || /^xlink:.+$/.exec(propertyName)) {
         return this.getObserver(obj, propertyName);
       }
-      if (/^\w+:|^data-|^aria-/.test(propertyName)
-        || obj instanceof DOM.SVGElement && this.svgAnalyzer.isStandardSvgAttribute(obj.nodeName, propertyName)) {
-        return dataAttributeAccessor;
-      }
+      return dataAttributeAccessor;
+      // if (/^\w+:|^data-|^aria-/.test(propertyName)
+      //   || obj instanceof DOM.SVGElement && this.svgAnalyzer.isStandardSvgAttribute(obj.nodeName, propertyName)) {
+      //   return dataAttributeAccessor;
+      // }
```

Comment:
Was this intentional? This would make every binding to a DOM element use attribute observation.

Reply:
The edited part belongs to `getAccessor` in observer locator, not `getObserver` so I think it's valid

Reply:
This would be a breaking change to getObserver. It's not needed for this PR is it?

Reply:
You meant return this back to previous state, and just keep the change in `dataAttributeAccessor` and `DataAttributeObserver` ? Isn't it same with before the fix, that `<a href.bind="prop"></a>` will still go to `propertyAccessor` ?

Reply:
Right, we'll need to make this more targeted. I think drastically changing the behavior here should wait until Aurelia vNext.

In the ObserverLocator's `createPropertyObserver` and `getAccessor` methods you'll need to add logic that looks for image.src, anchor.href, etc. Here's an [example](https://github.com/aurelia/binding/blob/master/src/observer-locator.js#L116).

Would be nice to remove all usages of `.toLowerCase()` in these conditions. Instead we could use instanceof or just compare with the uppercase string.

The logic in `createPropertyObserver` and `getAccessor` need to continue to match in terms of when element observers vs a property observer is used.

Reply:
What about we roll out previous fix for DOM special property (prop on the proto) only ? We can move `specialDataAttributeAccessor` part to `observer-locator.js` to and change the test to avoid exporting the new accessor ? It will fix the case `a.href='undefined'` and help avoid breaking change ?

```js
if (obj instance of DOM.Element) {
   ...
   if (/^\w+:|^data-|^aria-/.test(propertyName) || obj instanceof DOM.SVGElement && this.svgAnalyzer.isStandardSvgAttribute(obj.nodeName, propertyName)) {
        return dataAttributeAccessor;
   }
   ... the different is here, where we check if property name is in the element proto
   if (propertyName in obj) {
      return specialDomAttributeAccessor;
   }
}
... same after this point
```

Reply:
I think we can change `dataAttributeAccessor` and `DataAttributeObserver`. Currently setting null/undefined will result in the attribute value becoming the string `'undefined'`/`'null'` which I doubt anyone expects.

Reply:
Ok, so we will have some more checks like this:

```js
  if (/^\w+:|^data-|^aria-/.test(propertyName) || obj instanceof DOM.SVGElement && this.svgAnalyzer.isStandardSvgAttribute(obj.nodeName, propertyName)) {
        return dataAttributeAccessor;
  }
  if (propertyName === 'href' || propertyName === 'src') {
    return dataAttributeAccessor;
  }
```

Is this what you meant ?

Reply:
yeah